### DISCUSSION TITLE
Docker support

### DISCUSSION THREAD

**Discussion by bobyang3** [gnuton/asuswrt-merlin.ng#604] [Ideas]
Openwrt supports docker host now. I hope asuswrt will support docker container host as well. This will be very help to control home smart devices, set up proxy service...etc

https://openwrt.org/docs/guide-user/virtualization/docker_host

**Comment by gnuton** [comment]
How much memory do you need to run docker container? I see mich constraints
in the resources availability than a SW shortage

**Reply by bobyang3** [reply]
Good point. I am not sure, may be not that much? I think if openwrt has it then it should be ok because openwrt supports a lot of low end routers.

**Reply by gnuton** [reply]
docker requires 512 MB min and 2GB recommended btw.
Not sure it make sense. I ran it on raspberry pi 512 some time ago and it was painful and buggish.

**Reply by bobyang3** [reply]
Interesting. here is what I found for docker hoster in OpenWrt:

**Memory (RAM)**
The minimum recommended RAM for running Docker on OpenWrt is 128MB. However, for better performance and to run multiple containers, it's advisable to have at least 256MB or more of RAM.

**CPU**
A multi-core processor is recommended, but the exact requirements depend on the number and type of containers you plan to run. Generally, a dual-core processor with a clock speed of 800MHz or higher should be sufficient for basic Docker operations.

**Storage**
It's important to note that you'll need adequate storage space for Docker images and containers. A minimum of 4GB of storage is recommended, but more is better, especially if you plan to run multiple containers.
